template.go

```diff
@@ -29,52 +29,84 @@ func exists(path string) (bool, error) {
 	return false, err
 }
 
-func groupByMulti(entries []*RuntimeContainer, key, sep string) map[string][]*RuntimeContainer {
-	groups := make(map[string][]*RuntimeContainer)
-	for _, v := range entries {
-		value := deepGet(*v, key)
-		if value != nil {
-			items := strings.Split(value.(string), sep)
-			for _, item := range items {
-				groups[item] = append(groups[item], v)
-			}
+func getArrayValues(funcName string, entries interface{}) (reflect.Value, error) {
+	entriesVal := reflect.ValueOf(entries)
 
-		}
+	kind := entriesVal.Kind()
+
+	if kind == reflect.Ptr {
+		entriesVal = reflect.Indirect(entriesVal)
+		kind = entriesVal.Kind()
 	}
-	return groups
+
+	switch entriesVal.Kind() {
```

Comment:
You are right `kind` is better and reduces one call to `Kind()`. (It wont change the result because entriesVal is also updated when the pointer is dereferenced)

I think the issue #99 is already fixed with this PR and #100 is not needed.